Update troubleshooting.markdown

Zemoj opened the pull request:
Added media source fix for no TTS in YAML config.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Added troubleshooting text for a missing media source in YAML config.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->



## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Enhanced the troubleshooting guide for Assist by adding a new "Local Network Settings" section to assist users with complex network setups.
	- Added a "Missing Media Source" section to ensure users include the `media_source:` entry in their YAML configuration for proper functionality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Zemoj committed:
Update troubleshooting.markdown

Added media source fix for no TTS in YAML config.

frenck commented on source/voice_control/troubleshooting.markdown:
```diff
@@ -104,6 +104,15 @@ To fix the URL, follow these steps:
 
+### Missing Media Source
+
+If you are using YAML configuration make sure the media source is present:
```
This only applies if one isn't using the default_config integration (and thus has removed it manually). Let's make sure we mention that and reference it.

Zemoj committed:
Update troubleshooting.markdown

streamlined media source text and added refence for default config. added local network header for more consistent style.

frenck reviewed (changes requested):
Thanks, @Zemoj 👍 

../Frenck

frenck reviewed (approved):
Thanks, @Zemoj 👍 

../Frenck
